packages/neos-ui-editors/src/Editors/Button/index.js

```diff
@@ -0,0 +1,165 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Icon, IconButton} from '@neos-project/react-ui-components';
+import style from './style.module.css';
+import cx from 'classnames';
+import {neos} from '@neos-project/neos-ui-decorators';
+import {connect} from 'react-redux';
+import {selectors} from '@neos-project/neos-ui-redux-store';
+
+const getDataLoaderOptionsForProps = props => ({
+    contextNodePath: props.focusedNodePath,
+    dataSourceIdentifier: props.options.dataSourceIdentifier,
+    dataSourceUri: props.options.dataSourceUri,
+    dataSourceAdditionalData: props.options.dataSourceAdditionalData,
+    dataSourceDisableCaching: Boolean(props.options.dataSourceDisableCaching)
+});
+
+@neos(globalRegistry => ({
+    i18nRegistry: globalRegistry.get('i18n'),
+    dataSourcesDataLoader: globalRegistry.get('dataLoaders').get('DataSources')
+}))
+@connect(state => ({
+    focusedNodePath: selectors.CR.Nodes.focusedNodePathSelector(state)
+}))
+
+export default class ButtonEditor extends Component {
+    static propTypes = {
+        commit: PropTypes.func.isRequired,
+        highlight: PropTypes.bool,
+        value: PropTypes.any,
+        className: PropTypes.string,
+        options: PropTypes.shape({
+            allowEmpty: PropTypes.bool,
+            multiple: PropTypes.bool,
+            disabled: PropTypes.bool,
+            values: PropTypes.objectOf(PropTypes.shape({
+                label: PropTypes.string,
+                icon: PropTypes.string,
+                iconActive: PropTypes.string,
+                disabled: PropTypes.bool
+            })),
+
+            dataSourceIdentifier: PropTypes.string,
+            dataSourceUri: PropTypes.string,
+            dataSourceDisableCaching: PropTypes.bool,
+            dataSourceAdditionalData: PropTypes.objectOf(PropTypes.any)
+        }).isRequired,
+
+        i18nRegistry: PropTypes.object.isRequired
+    };
+
+    static defaultOptions = {
+        allowEmpty: true,
+        multiple: false,
+        disabled: false
+    };
+
+    state = {
+        selected: [],
+        isLoading: false,
+        buttons: []
+    };
+
+    options = {};
+
+    initialValueType = 'string';
+
+    constructor(props) {
+        super(props);
+        const {commit, options, value} = props;
+        this.options = Object.assign({}, this.constructor.defaultOptions, options)
+        // TODO: get actual property data type
```

Comment:
I've added that TODO line, to keep that in mind:
https://github.com/neos/neos-ui/pull/3475#discussion_r1239679057 
> AFAIK, the UI does not provide the configured type to the editors, but we should definitely change that. So hang on for a PR 😄, I'll see what I can do.